frontend/src/pages/modelServing/screens/projects/NIMServiceModal/NIMModelListSection.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+import { FormGroup } from '@patternfly/react-core';
+import { UpdateObjectAtPropAndValue } from '~/pages/projects/types';
+import { CreatingInferenceServiceObject, CreatingServingRuntimeObject } from '~/pages/modelServing/screens/types';
+import SimpleSelect from '~/components/SimpleSelect';
+import { fetchNIMModelNames, ModelInfo } from '~/pages/modelServing/screens/projects/utils';
+import { useDashboardNamespace } from '~/redux/selectors';
+
+type NIMModelListSectionProps = {
+  inferenceServiceData: CreatingInferenceServiceObject;
+  setInferenceServiceData: UpdateObjectAtPropAndValue<CreatingInferenceServiceObject>;
+  setServingRuntimeData: UpdateObjectAtPropAndValue<CreatingServingRuntimeObject>;
+  isEditing?: boolean;
+};
+
+const NIMModelListSection: React.FC<NIMModelListSectionProps> = ({
+  inferenceServiceData,
+  setInferenceServiceData,
+  setServingRuntimeData,
+  isEditing,
+}) => {
+  const [options, setOptions] = useState<{ key: string; label: string }[]>([]);
+  const [modelList, setModelList] = useState<ModelInfo[]>([]);
+  const { dashboardNamespace } = useDashboardNamespace();
+
+  useEffect(() => {
+    const getModelNames = async () => {
+      const modelInfos = await fetchNIMModelNames(dashboardNamespace);
+      if (modelInfos !== undefined) {
```

Comment:
![image](https://github.com/user-attachments/assets/e9e2d2c8-1e4e-4e3f-96ef-83a5208cfd89)

This is what happens... we should probably handle this case by at least disabling the dropdown and providing a message there was a problem with installation of NIMs or something like that.

The dropdown being enabled but not "open-able" is a problem (this is because of the lack of options).

Reply:
Updated.